pkg/local_object_storage/metabase/select.go

```diff
@@ -265,20 +273,75 @@ func (db *DB) selectFromFKBT(
 	f object.SearchFilter, // filter for operation and value
 	to map[string]int, // resulting cache
 	fNum int, // index of filter
-) { //
-	matchFunc, ok := db.matchers[f.Operation()]
-	if !ok {
-		db.log.Debug("missing matcher", zap.Uint32("operation", uint32(f.Operation())))
+) {
+	var nonNumMatcher matcher
+	op := f.Operation()
 
-		return
+	isNumOp := op == object.MatchNumGT || op == object.MatchNumGE || op == object.MatchNumLT || op == object.MatchNumLE
+	if !isNumOp {
+		var ok bool
+		nonNumMatcher, ok = db.matchers[f.Operation()]
+		if !ok {
+			db.log.Debug("missing matcher", zap.Uint32("operation", uint32(f.Operation())))
+			return
+		}
 	}
 
 	fkbtRoot := tx.Bucket(name)
 	if fkbtRoot == nil {
 		return
 	}
 
-	err := matchFunc.matchBucket(fkbtRoot, f.Header(), f.Value(), func(k, _ []byte) error {
+	if isNumOp {
+		// TODO: big math takes less code but inefficient
+		filterNum, ok := new(big.Int).SetString(f.Value(), 10)
+		if !ok {
+			db.log.Debug("unexpected non-decimal numeric filter", zap.String("value", f.Value()))
+			return
+		}
+
+		var objNum big.Int
+
+		err := fkbtRoot.ForEach(func(objVal, _ []byte) error {
```

Comment:
Slow. IIUC it can't be other way with the current DB scheme though.

Reply:
i didn't care much bout performance in initial implementation. Left #2757